diff --git a/haskell/private/context.bzl b/haskell/private/context.bzl
index d903ffa0..319fbafa 100644
--- a/haskell/private/context.bzl
+++ b/haskell/private/context.bzl
@@ -52,7 +52,7 @@ def haskell_context(ctx, attr = None):
         worker = worker,
         package_ids = package_ids,
         src_root = src_root,
-        package_root = ctx.label.workspace_root + ctx.label.package,
+        package_root = paths.join(ctx.label.workspace_root, ctx.label.package),
         env = env,
         mode = ctx.var["COMPILATION_MODE"],
         actions = ctx.actions,
