diff --git a/haskell/cabal.bzl b/haskell/cabal.bzl
index 6511d29f..9abdd67f 100644
--- a/haskell/cabal.bzl
+++ b/haskell/cabal.bzl
@@ -1689,6 +1689,8 @@ sh_binary(
     )
 
 def _stack_snapshot_impl(repository_ctx):
+    (os, _) = _get_platform(repository_ctx)
+
     snapshot = _parse_stack_snapshot(
         repository_ctx,
         repository_ctx.attr.snapshot,
@@ -1783,6 +1785,8 @@ load("@rules_haskell//haskell:defs.bzl", "haskell_library", "haskell_toolchain_l
 alias(name = "{name}", actual = "{actual}", visibility = {visibility})
 """.format(name = name, actual = vendored_packages[name], visibility = visibility),
             )
+        elif name == "unix" and os == "windows":
+            continue
         elif name in _CORE_PACKAGES:
             build_file_builder.append(
                 """
